server/user_test.go

```diff
@@ -0,0 +1,233 @@
+		{proto.Key("user2"), testUserConfig},
+		{proto.Key("user\xfe"), testUserConfig},
+	}
+
+	for _, test := range testData {
+		prefix := url.QueryEscape(string(test.prefix))
+		RunSetUser(testContext, prefix, testUserConfigBytes)
+		RunGetUser(testContext, prefix)
+	}
+	// Output:
+	// set user config for key prefix "user1"
+	// user config for key prefix "user1":
+	// hashed_password:
+	// - 10
+	// - 20
+	//
+	// set user config for key prefix "user2"
+	// user config for key prefix "user2":
+	// hashed_password:
+	// - 10
+	// - 20
+	//
+	// set user config for key prefix "user%FE"
+	// user config for key prefix "user%FE":
+	// hashed_password:
+	// - 10
+	// - 20
+}
+
+// Example_lsUsers creates a series of user configs and verifies
+// user-ls works. First, no regexp lists all user configs. Second,
+// regexp properly matches results.
+func Example_lsUsers() {
+	_, stopper := startAdminServer()
+	defer stopper.Stop()
+
+	keys := []proto.Key{
+		proto.Key("user1"),
+		proto.Key("user2"),
+		proto.Key("user3"),
+		proto.Key("foo"),
+	}
+
+	regexps := []string{
+		"",
+		"user*",
+		"user[12]",
+	}
+
+	for _, key := range keys {
+		prefix := url.QueryEscape(string(key))
+		RunSetUser(testContext, prefix, testUserConfigBytes)
+	}
+
+	for i, regexp := range regexps {
+		fmt.Fprintf(os.Stdout, "test case %d: %q\n", i, regexp)
+		if regexp == "" {
+			RunLsUser(testContext, "")
+		} else {
+			RunLsUser(testContext, regexp)
+		}
+	}
+	// Output:
+	// set user config for key prefix "user1"
+	// set user config for key prefix "user2"
+	// set user config for key prefix "user3"
+	// set user config for key prefix "foo"
+	// test case 0: ""
+	// [default]
+	// foo
+	// user1
+	// user2
+	// user3
+	// test case 1: "user*"
+	// user1
+	// user2
+	// user3
+	// test case 2: "user[12]"
+	// user1
+	// user2
+}
+
+// Example_rmUsers creates a series of user configs and verifies
+// user-rm works by deleting some and then all and verifying entries
+// have been removed via user-ls. Also verify the default user config
+// cannot be removed.
+func Example_rmUsers() {
+	_, stopper := startAdminServer()
+	defer stopper.Stop()
+
+	keys := []proto.Key{
+		proto.Key("user1"),
+		proto.Key("user2"),
+	}
+
+	for _, key := range keys {
+		prefix := url.QueryEscape(string(key))
+		RunSetUser(testContext, prefix, testUserConfigBytes)
+	}
+
+	for _, key := range keys {
+		prefix := url.QueryEscape(string(key))
+		RunRmUser(testContext, prefix)
+		RunLsUser(testContext, "")
+	}
+	// Output:
+	// set user config for key prefix "user1"
+	// set user config for key prefix "user2"
+	// removed user config for key prefix "user1"
+	// [default]
+	// user2
+	// removed user config for key prefix "user2"
+	// [default]
+}
+
+// Example_userContentTypes verifies that the Accept header can be used
+// to control the format of the response and the Content-Type header
+// can be used to specify the format of the request.
+func Example_userContentTypes() {
```

Comment:
I would be very happy to have all configs as part of a user-space table (as opposed to hard-coded system prefix).
However, I don't think we're quite there yet, and adding another one will not make the transition much more difficult. I'd say let's go with the existing format for now and revisit how accounting/perms/users/zones are stored later.